cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -327,24 +328,45 @@ pub enum TypedManifest {
 #[cfg_attr(test, derive(proptest_derive::Arbitrary))]
 #[serde(deny_unknown_fields)]
 pub struct TypedManifestCatalog {
-    pub(super) version: Version<1>,
+    pub version: Version<1>,
     /// The packages to install in the form of a map from install_id
     /// to package descriptor.
     #[serde(default)]
     pub install: ManifestInstall,
     /// Variables that are exported to the shell environment upon activation.
     #[serde(default)]
-    pub(super) vars: ManifestVariables,
+    pub vars: ManifestVariables,
     /// Hooks that are run at various times during the lifecycle of the manifest
     /// in a known shell environment.
     #[serde(default)]
-    pub(super) hook: ManifestHook,
+    pub hook: ManifestHook,
     /// Profile scripts that are run in the user's shell upon activation.
     #[serde(default)]
-    pub(super) profile: ManifestProfile,
+    pub profile: ManifestProfile,
     /// Options that control the behavior of the manifest.
     #[serde(default)]
     pub options: ManifestOptions,
+    /// Service definitions
+    #[serde(deserialize_with = "services_only_with_feature_flag")]
+    #[serde(default)]
+    #[serde(skip_serializing_if = "BTreeMap::is_empty")]
+    pub services: ManifestServices,
+}
+
+/// Only allow the manifest to deserialize a `services` section if the feature flag is present.
+///
+/// Note: generally it's a bad idea to look up feature flags outside of constructing the config
+///       at startup time, but we deserialize manifests in so many places that passing the config
+///       down to all of them isn't practical without a lot of refactoring.
+fn services_only_with_feature_flag<'de, D>(deserializer: D) -> Result<ManifestServices, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    if std::env::var(SERVICES_ENV_VAR).is_ok_and(|s| s.to_ascii_lowercase() == "true") {
+        Ok(ManifestServices(BTreeMap::deserialize(deserializer)?))
+    } else {
+        Err(D::Error::custom(ManifestError::ServicesNotEnabled))
+    }
 }
```

Comment:
making deserialization dependent on an env variable sourds wrong, because I'd usually assume serde operation to be pure functions.
In addition now we have to manage env vars in at least every test concerning services.

do we _need to_ deny this field?
why can't we silently ignore it when the feature flag isn't set?

the alternatives to this (e.g. [`DeserializeSeed`](https://docs.rs/serde/latest/serde/de/trait.DeserializeSeed.html)) or adding a struct validating method would unfortunately be quite wordy in comparison to this

Reply:
Discussed with Graham and we'll just silently ignore it

Reply:
Can you also add a test for the other variant where the services are present but the feature flag isn't?

Reply:
Since this field is ignored until it's used, I'm just ensuring that it parses no matter what (there's no interaction with the feature flag). The feature flag will come into play when you _need_ the `services` section.